maintenance/wikia/AddDiscussionsToNavigation.php

```diff
@@ -0,0 +1,135 @@
+ */
+class AddDiscussionsToNavigation extends Maintenance
+{
+
+	const VAR_NAME = 'wgOasisGlobalNavigation';
+	const NAVIGATION_ELEMENT = '**Special:DiscussionsNavigation|Discussions';
+	const FORUM_NAVIGATION_ELEMENT = '**Special:Forum|Forum';
+	const REASON = 'SOC-2816';
+	const SPLIT = '\n';
+
+	private $dryRun = false;
+	private $verbose = false;
+	private $rollback = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Update " . self::VAR_NAME . " with Discussions navigation "
+			. self::NAVIGATION_ELEMENT;
+		$this->addOption( 'dry-run', 'Dry run mode', false, false, 'd' );
+		$this->addOption( 'verbose', 'Show extra debugging output', false, false, 'v' );
+		$this->addOption( 'rollback', 'Rollback the operation', false, false, 'r' );
+	}
+
+	public function execute() {
+		$this->dryRun = $this->hasOption( 'dry-run' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->rollback = $this->hasOption( 'verbose' );
+
+
+		$total = 0;
+		$successful = 0;
+		foreach ($this->getAllSiteIds() as $siteId) {
+			$success = $this->updateSiteId(trim($siteId));
+			if ($success) {
+				$successful++;
+			}
+			$total++;
+		}
+
+		echo "\nTotal sites: $total, Success: {$successful}, Failed: " . ($total - $successful) . "\n\n";
+	}
+
+	private function getAllSiteIds() {
+		return [3035];
+	}
+
+	private function updateSiteId($siteId) {
+		$varData = (array)WikiFactory::getVarByName( self::VAR_NAME, $siteId, true );
+		if ( empty( $varData['cv_id'] ) ) {
+			$this->debug("Error:" . self::VAR_NAME . " not found.\n");
+		}
+
+		if ( $varData['cv_variable_type'] !== 'text' ) {
+			echo $varData['cv_variable_type'];
+			die("Error: " . self::VAR_NAME . " is not a text and the script cannot append to it anything.\n");
+		}
+
+		echo "Variable: " . self::VAR_NAME . " (Id: $varData[cv_id])\n";
+		$this->debug( "Variable data: " . json_encode( $varData ) . "\n");
+
+		$value = $this->createNewValue( unserialize( $varData[ 'cv_value' ] ) );
+
+		$result = false;
+		if ( !$this->dryRun ) {
+//			$this->setVariable( $siteId, $value, self::REASON );
+		}
+		return $result;
+	}
+
+	private function createNewValue( $previousValue ) {
+		$this->debug( "Previous value " . $previousValue );
+		$array = explode( self::SPLIT, $previousValue );
+		if ( !$this->rollback ) {
+			$this->insertAfterForumOrAtTheEnd( $array );
+		} else {
+			$this->removeNavigationElement( $array );
+		}
+		$newValue = join( self::SPLIT, $array );
+		$this->debug( "New value " . $previousValue );
+		return $newValue;
+	}
+
+	function insertAfterForumOrAtTheEnd( &$array ) {
+		$pos = array_search( self::FORUM_NAVIGATION_ELEMENT, $array );
+		if ( $pos === false ) {
+			$pos = count( $array );
+			$this->debug( "Forum not found - appending at the end." );
+		} else {
+			$this->debug( "Found forum at $pos position. ");
+		}
+		$array = array_merge(
+			array_slice( $array, 0, $pos ),
+			self::NAVIGATION_ELEMENT,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)